Firestore/core/src/local/leveldb_remote_document_cache.cc

```diff
@@ -170,115 +171,82 @@ MutableDocumentMap LevelDbRemoteDocumentCache::GetAll(
 }
 
 MutableDocumentMap LevelDbRemoteDocumentCache::GetAllExisting(
-    const DocumentKeySet& keys) {
-  MutableDocumentMap docs = LevelDbRemoteDocumentCache::GetAll(keys);
+    const DocumentVersionMap key_version_map) const {
   MutableDocumentMap result;
-  for (const auto& kv : docs) {
-    const DocumentKey& key = kv.first;
-    auto& document = kv.second;
+  for (const auto& pair : key_version_map) {
+    auto document = Get(pair.first).WithReadTime(pair.second);
```

Comment:
We probably should use the background queue to do the deserialization: see line 144 for an example.